crates/binstalk/src/helpers/remote.rs

```diff
@@ -53,22 +64,47 @@ impl Client {
         &self.client
     }
 
+    async fn send_request_inner(
+        &self,
+        method: &Method,
+        url: &Url,
+    ) -> Result<Response, reqwest::Error> {
+        loop {
```

Comment:
We probably want an iteration limit here, to avoid infinitely waiting...

Reply:
Done

Reply:
I've added an iteration limit of 3 here.